File: frontend/types/isp_customer.ts
export interface ISPCustomer {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  username: string;
  organization: {
    id: string;
    name: string;
  };
  package: {
    id: string;
    name: string;
  };
  station: {
    id: string;
    name: string;
  };
  expirationDate: string;
  status: "ACTIVE" | "INACTIVE";
  online: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface ISPCustomersResponse {
  customers: {
    success: boolean;
    message: string;
    customers: ISPCustomer[];
  };
}

export interface CustomerResponse {
  customer: {
    __typename: string;
    id: string;
    firstName: string;
    lastName: string;
    username: string;
    status: string;
    // ... other fields
  };
}

export interface CreateISPCustomerInput {
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  username: string;
  password: string;
  organizationId: string;
  packageId: string;
  stationId: string;
  expirationDate: string;
}

export interface UpdateISPCustomerInput {
  firstName?: string;
  lastName?: string;
  email?: string;
  phone?: string;
  username?: string;
  password?: string;
  packageId?: string;
  stationId?: string;
  expirationDate?: string;
  status?: "ACTIVE" | "INACTIVE";
} 

File: backend/app/models/isp_customer.py
from datetime import datetime
from typing import Optional, List
from app.schemas.enums import IspManagerCustomerStatus
import strawberry

@strawberry.type
class DBISPCustomer:
    _id: str
    firstName: str
    lastName: str
    email: str
    phone: str
    username: str
    password: str
    organizationId: str
    packageId: str
    stationId: str
    expirationDate: datetime
    status: IspManagerCustomerStatus
    online: bool
    createdAt: datetime
    updatedAt: datetime
    reminderDaysSent: Optional[List[int]] = None